test/QueryToSocketDistributorTest.cpp

```diff
@@ -25,21 +25,25 @@ static constexpr auto noop = ad_utility::noop;
 #define return co_return
 
 ASYNC_TEST(QueryToSocketDistributor, addQueryStatusUpdateThrowsWhenFinished) {
-  QueryToSocketDistributor queryToSocketDistributor{ioContext, noop};
-  co_await queryToSocketDistributor.signalEnd();
-  EXPECT_THROW(co_await queryToSocketDistributor.addQueryStatusUpdate("Abc"),
+  auto ptr = std::make_shared<QueryToSocketDistributor>(ioContext, noop);
+  QueryToSocketDistributor& queryToSocketDistributor = *ptr;
+  queryToSocketDistributor.signalEnd();
+  EXPECT_THROW(queryToSocketDistributor.addQueryStatusUpdate("Abc"),
                ad_utility::Exception);
+  co_return;
```

Comment:
Also same point here, if there's just co_return then it should most likely be a `TEST` instead of `ASYNC_TEST`